base/reflectx/structs.go

```diff
@@ -229,3 +230,110 @@ func StructTags(tags reflect.StructTag) map[string]string {
 func StringJSON(v any) string {
 	return string(errors.Log1(jsonx.WriteBytesIndent(v)))
 }
+
+// FieldValue returns the [reflect.Value] of given field within given struct value,
+// where the field can be a path with . separators, for fields within struct fields.
+func FieldValue(s reflect.Value, fieldPath string) (reflect.Value, error) {
+	sv := UnderlyingPointer(s)
+	var zv reflect.Value
+	if sv.Elem().Kind() != reflect.Struct {
+		return zv, errors.New("reflectx.FieldValue: kind is not struct")
+	}
+	fps := strings.Split(fieldPath, ".")
+	fv := sv.Elem().FieldByName(fps[0])
+	if fv == zv {
+		return zv, errors.New("reflectx.FieldValue: field name not found: " + fps[0])
+	}
+	if len(fps) == 1 {
+		return fv, nil
+	}
+	return FieldValue(fv, strings.Join(fps[1:], "."))
```

Comment:
This would be much cleaner as a for loop instead of a recursive algorithm that splits and joins the path over and over again... Also, just do `Underlying` instead of getting `Elem()` every time.